app/lib/common/providers/notifiers/attachments_notifiers.dart

```diff
@@ -36,3 +42,77 @@ class AttachmentsManagerNotifier
     return arg;
   }
 }
+
+class AttachmentDraftsNotifier extends StateNotifier<List<AttachmentDraft>> {
+  final AttachmentsManager manager;
+  final Ref ref;
+
+  AttachmentDraftsNotifier({required this.manager, required this.ref})
+      : super([]);
+
+  /// converts user selected media to attachment draft and adds to state list.
+  /// only supports image/video/audio/file.
+  Future<void> addDraft(File file) async {
+    EasyLoading.show(status: 'Adding draft');
+    final client = ref.read(alwaysClientProvider);
+    try {
+      final mimeType = lookupMimeType(file.path)!;
+      if (mimeType.startsWith('image/')) {
+        Uint8List bytes = await file.readAsBytes();
+        final decodedImage = await decodeImageFromList(bytes);
```

Comment:
this code looks warily similar to the one for chat attachment sending below, why is that not generalized somewhere?!?

Reply:
Yeah this keeps duplicating, will add utility fn for it.

Reply:
as said above, let's revert the chat code and keep this separated for now in this PR. Let's add a tech-debt issue that we want to address these, but not bloat up this PR even more...